File: life-os/apps/api/src/main/java/tech/buildwithpartha/lifeos/project/domain/Milestone.java
package tech.buildwithpartha.lifeos.project.domain;

import java.time.Instant;
import java.time.LocalDate;
import java.util.Objects;
import java.util.Optional;
import java.util.UUID;

/** Immutable domain record representing a Milestone checkpoint. */
public record Milestone(
    UUID id,
    UUID projectId,
    String title,
    Optional<LocalDate> date,
    MilestoneStatus status,
    int ordering,
    Instant createdAt,
    Instant updatedAt,
    long version) {
  public Milestone {
    Objects.requireNonNull(id, "id must not be null");
    Objects.requireNonNull(projectId, "projectId must not be null");
    Objects.requireNonNull(title, "title must not be null");
    Objects.requireNonNull(date, "date must not be null");
    Objects.requireNonNull(status, "status must not be null");
    Objects.requireNonNull(createdAt, "createdAt must not be null");
    Objects.requireNonNull(updatedAt, "updatedAt must not be null");
  }
}

File: life-os/apps/web/src/features/projects/model/milestone.ts
/**
 * Milestone domain models (LOS-0704, LOS-0713).
 */

export type MilestoneStatus = "PLANNED" | "COMPLETED" | "CANCELLED";

export interface Milestone {
  readonly id: string;
  readonly projectId: string;
  readonly title: string;
  readonly date?: string | null; // LocalDate format: YYYY-MM-DD
  readonly status: MilestoneStatus;
  readonly ordering: number;
  readonly createdAt: string; // ISO DateTime
  readonly updatedAt: string; // ISO DateTime
  readonly version: number;
}
